### ISSUE TITLE
Namespacing redis entries - Design Consideration

### ISSUE THREAD

**Issue by ScheerMT** [dudleycarr/ratelimit.js#15]
This would be a design decision to be made.. right now, I don't see a method to have the rate limiter namespace their entries to allow for multiple servers running this library to work within the same redis db without some kind of terrible collisions. 

A possible work around I have thought of for the immediate fix would be to start a different redis connection and point it to a seperate db for each server. This is less than ideal because of the maintence of "What the heck is db2 associated with again?" instead of looking at the keys themselves and seeing the name associated with the whitelist items or even rate limited IP's!

**EDIT**: Decided to go back and have a look through the source again..appears you are able to make a custom namespace for the limiting keys but not for the whitelist or blacklist... It seems that the names are hard coded in two places.. lua scripts and then again in the rate limit class.

**Comment by joshgummersall** [comment]
You can actually specify a prefix in the `RateLimit` constructor as the third parameter. [The constructor is here](https://github.com/dudleycarr/ratelimit.js/blob/master/src/rate_limit.coffee#L20) and you can see that the third parameter is the prefix (which defaults to "ratelimit" if it is not provided).

**Comment by joshgummersall** [comment]
You are right about the Whitelist keys being hardcoded. That is something that could be worth revisiting, although a bit more difficult as it involves passing more configuration to the Lua scripts through Redis.

**Comment by ScheerMT** [comment]
Depending on the library you are using for redis, it may have namespace prefixing built in.. Like [ioredis](https://github.com/luin/ioredis#transparent-key-prefixing). 

**HOWEVER**, I tested this and the ratelimiting whitelist breaks when using a namespace prefix from this redis client. The key is created but is not referenced correctly when trying to allow requests to come through. I remove the prefix from the redis client and the functionality for whitelisted IP's works as expected. 

Between tests i did a FLUSHDB to ensure a clean starting point between each test. I think this feature would still be worthwhile to extend the prefix to the whitelist and blacklist to clear up the functionality expectations from this library independent of others (including redis)

tl;dr - the prefix functionality of ioredis doesn't work for whitelist in this library probably due to hard coded lua script names

**Comment by joshgummersall** [comment]
@ScheerMT yes, agreed. I'll work on a fix for this.

**Comment by ScheerMT** [comment]
Once this fix is in place, I'd consider adding the prefix functionality to the README documentation..
